pipeline/transformer/transformer_sierra/src/test/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraMeetingSubjectsTest.scala

```diff
@@ -0,0 +1,120 @@
+    with SierraDataGenerators {
+
+  def bibId = createSierraBibNumber
+
+  def bibData(varFields: VarField*) =
+    createSierraBibDataWith(varFields = varFields.toList)
+
+  def varField(tag: String, subfields: MarcSubfield*) =
+    createVarFieldWith(
+      marcTag = tag,
+      subfields = subfields.toList,
+      indicator2 = "0")
+
+  it("returns zero subjects if there are none") {
+    SierraMeetingSubjects(bibId, bibData()) shouldBe Nil
+  }
+
+  it("returns subjects for varfield 611, subfield $$a") {
+    val data = bibData(
+      varField("600", MarcSubfield(tag = "a", content = "Not content")),
+      varField("611", MarcSubfield(tag = "a", content = "Content")),
+    )
+    SierraMeetingSubjects(bibId, data) shouldBe List(
+      Unidentifiable(
+        Subject(
+          label = "Content",
+          concepts = List(Unidentifiable(Meeting(label = "Content")))
+        )
+      )
+    )
+  }
+
+  it("returns subjects with subfields $$a, $$c and $$d concatenated in order") {
+    val data = bibData(
+      varField(
+        "611",
+        MarcSubfield(tag = "c", content = "C"),
+        MarcSubfield(tag = "a", content = "A"),
+        MarcSubfield(tag = "d", content = "D"),
+      )
+    )
+    SierraMeetingSubjects(bibId, data) shouldBe List(
+      Unidentifiable(
+        Subject(
+          label = "A C D",
+          concepts = List(Unidentifiable(Meeting(label = "A C D")))
+        )
+      )
+    )
+  }
+
+  it("returns zero subjects if subfields all $$a, $$c and $$d are missing") {
+    val data = bibData(
+      varField("611", MarcSubfield(tag = "x", content = "Hmmm"))
+    )
+    SierraMeetingSubjects(bibId, data) shouldBe Nil
+  }
+
+  it("creates an identifiable subject if subfield 0") {
+    val data = bibData(
+      varField(
+        "600",
+        MarcSubfield(tag = "a", content = "Content"),
+        MarcSubfield(tag = "0", content = "lcsh7212")
+      )
+    )
+    val sourceIdentifier = SourceIdentifier(
+      identifierType = IdentifierType("lc-names"),
+      ontologyType = "Subject",
+      value = "lcsh7212"
+    )
+    SierraPersonSubjects(bibId, data) shouldBe List(
+      Identifiable(
+        Subject(
+          label = "Content",
+          concepts = List(Unidentifiable(Person(label = "Content")))
+        ),
+        sourceIdentifier = sourceIdentifier
+      )
+    )
+  }
+
+  it("returns multiple subjects if multiple 611") {
+    val data = bibData(
+      varField("611", MarcSubfield(tag = "a", content = "First")),
+      varField("611", MarcSubfield(tag = "a", content = "Second")),
+    )
+    SierraMeetingSubjects(bibId, data) shouldBe List(
+      Unidentifiable(
+        Subject(
+          label = "First",
+          concepts = List(Unidentifiable(Meeting(label = "First")))
+        )
+      ),
+      Unidentifiable(
+        Subject(
+          label = "Second",
+          concepts = List(Unidentifiable(Meeting(label = "Second")))
+        )
+      )
+    )
+  }
+}
```

Comment:
We could wrap most these tests up into 1 table driven test.
If we're worried about the logging, we should probably try to fix the logging with the table test libs we use.

Reply:
Agree that these tests maybe a bit verbose, but still not convinced that table driven tests are the way to go. Each of the tests are actually testing different things, so even if the structure of the tests is similar I reckon makes sense to have separate unit tests.